src/D2L.Bmx/UpdateHandler.cs

```diff
@@ -0,0 +1,130 @@
+using System.Formats.Tar;
+using System.IO.Compression;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace D2L.Bmx;
+
+internal class UpdateHandler {
+
+	public async Task HandleAsync() {
+		using var httpClient = new HttpClient();
+		var releaseData = await GithubUtilities.GetLatestReleaseDataAsync();
+		var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+		var latestVersion = new Version( GithubUtilities.GetReleaseVersion( releaseData ) );
+		if( latestVersion <= localVersion ) {
+			Console.WriteLine( $"Already own the latest version {latestVersion}" );
+			return;
+		}
+		string archiveName = GetOSFileName();
+		string downloadUrl = releaseData?.Assets?.FirstOrDefault( a => a.Name == archiveName )?.BrowserDownloadUrl
+			?? string.Empty;
```

Comment:
we check null or whitespace below, so there's no reason at all to fall back to empty string here
```suggestion
		string? downloadUrl = releaseData?.Assets?.FirstOrDefault( a => a.Name == archiveName )?.BrowserDownloadUrl;
```